### PR TITLE
Unify Streaming Services

### PR DESCRIPTION
**Pull request by prmukherj**
An attempt to unify all the streaming services like transcript, events_manager, monitor_threads, etc.

### CODE REVIEW

File: src/ansys/fluent/core/streaming_services/streaming_services.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            ID of the registered callback.
+        """
+        with self._lock:
+            if callback_id in self._service_callbacks:
+                del self._service_callbacks[callback_id]
+
+    def start(self) -> None:
+        """Start streaming of Fluent transcript."""
+        with self._lock:
+            if self._stream_thread is None:
```
Comment (ajain-work) [inline src/ansys/fluent/core/streaming_services/streaming_services.py:57]: not is_streaming

File: src/ansys/fluent/core/streaming_services/monitor_manager_service.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                if df.empty
+                else (
+                    df.index.to_numpy(),
+                    {column: df[[column]].to_numpy() for column in df.columns},
+                )
+            )
+
+    def register_on_monitor_refresh_callback(
+        self, on_monitor_refresh_callback: Callable[[], Any]
+    ):
```
Comment (ajain-work) [inline src/ansys/fluent/core/streaming_services/monitor_manager_service.py:124]: why? We can remove it now

Comment (prmukherj) [reply src/ansys/fluent/core/streaming_services/monitor_manager_service.py]: Done, Thank you

File: src/ansys/fluent/core/streaming_services/streaming_services.py (hunk 1 above)
Comment (prmukherj) [reply src/ansys/fluent/core/streaming_services/streaming_services.py]: Done, Thank you

**Review by seanpearsonuk — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Review by ajain-work — APPROVED** [review]

**Pull request merged**